lib/build.gradle.kts

```diff
@@ -70,7 +70,22 @@ val buildCrypto = task<Exec>("BuildCrypto") {
 }
 
 val compileContracts = task<Exec>("compileContracts") {
-    commandLine("${project.projectDir}/src/test/resources/compileContracts.sh")
+    val compileScriptPath = "${project.projectDir}/src/test/resources/compileContracts.sh"
+
+    val sourceDirectoryV0 = "${project.projectDir}/src/test/resources/src_v0"
+    val sourceDirectoryV1 = "${project.projectDir}/src/test/resources/src_v1"
+
+    val compileDirectoryV0 = layout.buildDirectory.dir("lib/src/test/resources/compiled_v0")
+    val compileDirectoryV1 = layout.buildDirectory.dir("lib/src/test/resources/compiled_v1")
+
+    inputs.file(compileScriptPath)
+    inputs.dir(sourceDirectoryV0)
+    inputs.dir(sourceDirectoryV1)
+
+    outputs.dir(compileDirectoryV0)
+    outputs.dir(compileDirectoryV1)
+
+    commandLine(compileScriptPath)
```

Comment:
what does this do?

Reply:
Sets input and output for compileContracts task. This allows to locally cache compiled contracts. If there are no changes to any files in input dirs, script and output dirs, the task will not recompile contracts.

Reply:
But we already have something like this in compileContracts.sh right? If contracta are built, the scripts skips this step.

Reply:
No, we don't. We check whether compilers are built, but we recompile contracts each time.

Reply:
I'm not sure it uses checksums per se, but it is based on both metadata and file contents and works as expected: https://docs.gradle.org/current/userguide/incremental_build.html#sec:task_inputs_outputs